src/serde/de.rs

```diff
@@ -410,11 +417,26 @@ impl<'a, 'de> de::Deserializer<'de> for ValueDeserializer<'a, 'de> {
     where
         V: Visitor<'de>,
     {
-        // annotations are currently only supported with parameterized Rust enums
-        if let Some(annotation) = self.value.annotations().next() {
-            visitor.visit_str(annotation?.text().unwrap())
-        } else {
-            visitor.visit_str(self.value.read()?.expect_text()?)
+        let mut annotations = self.value.annotations();
+        let first_annotation = annotations.next().transpose()?;
+        let second_annotation = annotations.next().transpose()?;
+        match (first_annotation, second_annotation) {
```

Comment:
If I understand this correctly, it will fail if you include _only_ the enum variant annotation. While I could be convinced that it's okay to always write the type annotation, I don't think we should require a type annotation on read because it makes this harder to use it with object mappers in other language ecosystems.

Reply:
I think I'll take your original suggestion and remove the type annotations that I added in this PR. I'm going to do that after my next PR lands though.